### DISCUSSION TITLE
This is a bug? [static-embed-files example]

### DISCUSSION THREAD

**Discussion by josejachuf** [salvo-rs/salvo#153] [Q&A] [answered]
Hi 

This is a bug or am I doing something wrong?

If I clone the Salvo repository and run static-embed-files, it works fine, but if I try to use the same code outside, it fails. Obviously I changed the Salvo dependence in cargo.toml

main.rs

```
use rust_embed::RustEmbed;
use salvo::prelude::*;

#[derive(RustEmbed)]
#[folder = "static"]
struct Assets;

#[tokio::main]
async fn main() {
    tracing_subscriber::fmt().init();

    let router = Router::with_path("<**path>").get(static_embed::<Assets>().with_fallback("index.html"));
    tracing::info!("Listening on http://127.0.0.1:7878");
    Server::new(TcpListener::bind("127.0.0.1:7878")).serve(router).await;
}

```

cargo.toml

```
[dependencies]
# askama = "0.11"
rust-embed = "6"
salvo = { version = "0.35", features = ["serve-static"] }
tokio = { version = "1", features = ["macros"] }
tracing = "0.1"
tracing-subscriber = "0.3"
```

> error[E0433]: failed to resolve: use of undeclared crate or module `tracing`
>    --> /.cargo/registry/src/github.com-1ecc6299db9ec823/salvo_extra-0.35.0/src/serve_static/dir.rs:218:33
>     |
> 218 | ...                   tracing::error!(error = ?e, "redirect failed");
>     |                       ^^^^^^^ use of undeclared crate or module `tracing`

**Comment by chrislearn** [comment, marked as answer]
It's bug. I have published 0.35.1, can you checked is it fixed?

**Reply by josejachuf** [reply]
This works  fine, now compile. I have not tried much more than that

**Reply by josejachuf** [reply]
Thanks!
